src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -440,38 +387,173 @@ impl<S> DdlServiceImpl<S>
 where
     S: MetaStore,
 {
-    async fn create_mview_on_compute_node(
+    fn get_dependent_relations(&self, fragment_graph: &StreamFragmentGraph) -> RwResult<Vec<u32>> {
+        // TODO: distinguish SourceId and TableId
+        fn resolve_dependent_relations(
+            stream_node: &StreamNode,
+            dependent_relations: &mut HashSet<TableId>,
+        ) -> RwResult<()> {
+            match stream_node.node_body.as_ref().unwrap() {
+                NodeBody::Source(source_node) => {
+                    dependent_relations.insert(source_node.get_table_id());
+                }
+                NodeBody::Chain(chain_node) => {
+                    dependent_relations.insert(chain_node.get_table_id());
+                }
+                _ => {}
+            }
+            for child in &stream_node.input {
+                resolve_dependent_relations(child, dependent_relations)?;
+            }
+            Ok(())
+        }
+
+        let mut dependent_relations = Default::default();
+        for fragment in fragment_graph.fragments.values() {
+            resolve_dependent_relations(fragment.node.as_ref().unwrap(), &mut dependent_relations)
+                .map_err(tonic_err)?;
+        }
+        Ok(dependent_relations.into_iter().collect())
+    }
+
+    fn get_internal_table(&self, ctx: &CreateMaterializedViewContext) -> RwResult<Vec<Table>> {
```

Comment:
Ditto.

Reply:
`fn get_internal_table` needs `self` to access `fn set_table_mapping`. I could change it's signature by adding an argument but that seems unnecessary so I rather not touch it.